### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by mikedonnici**

### CODE REVIEW

File: geometry.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+type rect struct {
+	width, height float64
+}
+type circle struct {
+	radius float64
+}
+
+func (r rect) area() float64 {
```
Comment (albertsundjaja) [inline geometry.go:19]: although we are not modifying the member variable, it might be beneficial to use a pointer receiver here to improve performance

File: geometry_test.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package geometry
+
+import (
+	"math"
+	"testing"
+)
+
+func TestRect(t *testing.T) {
```
Comment (albertsundjaja) [inline geometry_test.go:8]: since width and height are of type float. We might also want to have more test to cover more edge cases

More edge cases:
1. when width or height are 0
2. when width or height are negative
3. when width or height are float

File: geometry.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	"math"
+)
+
+type geometry interface {
+	area() float64
+	perim() float64
+}
+
+type rect struct {
+	width, height float64
```
Comment (albertsundjaja) [inline geometry.go:13]: we might want to use a getter / setter to set width/height/radius, as it does not make sense to have width, height, radius <= 0

File: geometry_test.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		t.Fatalf("area() = %f, want 40", r.area())
+	}
+	if r.perim() != 26 {
+		t.Fatalf("perim() = %f, want 26", r.perim())
+	}
+}
+
+func TestCircle(t *testing.T) {
+	c := circle{
+		radius: 10,
```
Comment (albertsundjaja) [inline geometry_test.go:23]: same comment as for rectangles, more test to cover more edge cases

File: geometry_test.go [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+	}
+	wantArea := math.Round(314.159265)
```
Comment (albertsundjaja) [inline geometry_test.go:25]: this will round to nearest integer. Suggestion is to define a precision point
